src/arrow2/src/array/map/mod.rs

```diff
@@ -195,6 +202,43 @@ impl MapArray {
 impl Array for MapArray {
     impl_common_array!();
 
+    fn convert_logical_type(&self, target: DataType) -> Box<dyn Array> {
+        tracing::trace!("converting logical type to\n{target:#?}");
+        let outer_is_map = matches!(target, DataType::Map { .. });
+
+        if outer_is_map {
+            // we can do simple conversion
```

Comment:
Nit: I think a more useful comment would be explaining why we can do this simple conversion, instead of stating that it's a simple conversion before doing a simple conversion.

Reply:
321c3510a243630a27c8e0073833af2e9769ebaf